src/Raven.Server/Documents/RemoteAttachmentsSender.cs

```diff
@@ -130,132 +135,126 @@ internal async Task<int> ProcessRemoteAttachments(int batchSize, long maxItemsTo
+                                            if (attachment.RemoteParameters is null)
+                                                continue;
 
-                                            states.Enqueue(document);
+                                            if (currentTime < attachment.RemoteParameters.At)
+                                                continue;
 
-                                            continue;
-                                        }
+                                            var identifier = attachment.RemoteParameters.Identifier;
 
-                                        // add duplicates queue for this document which is being processed now
-                                        docsStates.Add(document.Id, new Queue<AttachmentRemoteInfo>());
+                                            if (Configuration.Destinations.TryGetValue(identifier, out var destination) == false || destination.HasUploader() == false)
+                                            {
+                                                // destination no longer exist or is disabled, skip uploading this attachment
+                                                HandleSkippedItem(document.Id, attachment, destination);
+                                                continue;
+                                            }
 
-                                        var index = Task.WaitAny(uploadTasks);
-                                        try
-                                        {
-                                            var res = await uploadTasks[index];
-                                            HandleUploadTaskResult(res, processed, docsStates);
-                                        }
-                                        catch (Exception e)
-                                        {
-                                            _exceptions.Add(e);
-                                        }
-                                        finally
-                                        {
-                                            uploadTasks[index] = CreateUploadTaskAsync(directUploaders, currentTime, document);
+                                            attachmentsToUploadByIdentifier
+                                                 .GetOrAdd(identifier)
+                                                 .GetOrAdd(attachment.Base64Hash.ToString())
+                                                 .DocumentIds.Add(document.Id);
+
+                                            if (directUploaders.ContainsKey(identifier) == false)
+                                            {
+                                                directUploaders[identifier] = new AttachmentUploader(UploaderSettings.GenerateDirectUploaderSettingsForAttachments(_database, identifier, destination.S3Settings, destination.AzureSettings), Logger, _token);
```

Comment:
directUploaders does not have case insensitive comparer on the identifier. Can you confirm that this is OK?